src/it/java/teammates/it/storage/sqlapi/UsersDbIT.java

```diff
@@ -276,4 +286,301 @@ public void testGetStudentsByGoogleId()
+    }
+
+    @Test
+    public void testSqlInjectionInCreateStudent() throws Exception {
+        ______TS("SQL Injection test in createStudent email field");
+
+        String email = "test';/**/DROP/**/TABLE/**/users;/**/--@gmail.com";
+        Student studentEmail = getTypicalStudent();
+        studentEmail.setEmail(email);
+
+        // The regex check should fail and throw an exception
+        assertThrows(InvalidParametersException.class,
+                () -> usersDb.createStudent(studentEmail));
+
+        ______TS("SQL Injection test in createStudent name field");
+        Student studentName = getTypicalStudent();
+        studentName.setEmail("ins.usersdbit.3@gmail.com");
+        String name = "test';/**/DROP/**/TABLE/**/accounts;/**/--";
+        studentName.setName(name);
+        String studentNameRegKey = "ins.usersdbit.regkey3";
+        studentName.setRegKey(studentNameRegKey);
+
+        usersDb.createStudent(studentName);
+
+        HibernateUtil.flushSession();
+
+        // The system should treat the input as a plain text string
+        Student actualStudent = usersDb.getStudentByRegKey(studentNameRegKey);
+        assertEquals(actualStudent.getName(), name);
+    }
+
+    @Test
+    public void testSqlInjectionInGetInstructorByRegKey() throws Exception {
+        ______TS("SQL Injection test in getInstructorByRegKey");
+
+        Instructor instructor = getTypicalInstructor();
+        instructor.setEmail("instructorregkey.usersdbit@gmail.com");
+
+        usersDb.createInstructor(instructor);
+
+        // The system should treat the input as a plain text string
+        String regKey = "test' OR 1 = 1; --";
+        Instructor actualInstructor = usersDb.getInstructorByRegKey(regKey);
+        assertNull(actualInstructor);
+    }
+
+    @Test
+    public void testSqlInjectionInGetInstructorByGoogleId() throws Exception {
+        ______TS("SQL Injection test in getInstructorByGoogleId courseId field");
+        String injection = "test' OR 1 = 1; --";
+        assertNull(usersDb.getInstructorByGoogleId(injection, instructor.getAccount().getGoogleId()));
+
+        ______TS("SQL Injection test in getInstructorByGoogleId googleId field");
+        assertNull(usersDb.getInstructorByGoogleId(instructor.getCourseId(), injection));
+    }
+
+    @Test
+    public void testSqlInjectionInGetInstructorsDisplayedToStudents() throws Exception {
+        ______TS("SQL Injection test in getInstructorsDisplayedToStudents courseId field");
+        String injection = "test' OR 1 = 1; --";
+        assertEquals(usersDb.getInstructorsDisplayedToStudents(injection).size(), 0);
+    }
+
+    @Test
+    public void testSqlInjectionInGetStudentByRegKey() throws Exception {
+        ______TS("SQL Injection test in getStudentByRegKey");
+
+        Student student = getTypicalStudent();
+        student.setEmail("studentregkey.usersdbit@gmail.com");
+
+        usersDb.createInstructor(instructor);
```

Comment:
It seems unnecessary to create an instructor here.

Reply:
ah oops, I was meant to create a student here, changed